packages/trpc/server/routers/viewer/webhook.tsx

```diff
@@ -19,62 +19,65 @@ const webhookIdAndEventTypeIdSchema = z.object({
   eventTypeId: z.number().optional(),
 });
 
-export const webhookRouter = createProtectedRouter()
-  .middleware(async ({ ctx, rawInput, next }) => {
-    // Endpoints that just read the logged in user's data - like 'list' don't necessary have any input
-    if (!rawInput) {
-      return next();
-    }
-    const webhookIdAndEventTypeId = webhookIdAndEventTypeIdSchema.safeParse(rawInput);
-    if (!webhookIdAndEventTypeId.success) {
-      throw new TRPCError({ code: "PARSE_ERROR" });
-    }
-    const { eventTypeId, id } = webhookIdAndEventTypeId.data;
+const webhookProcedure = authedProcedure.use(async ({ ctx, rawInput, next }) => {
+  // Endpoints that just read the logged in user's data - like 'list' don't necessary have any input
+  if (!rawInput) {
+    return next();
+  }
+  const webhookIdAndEventTypeId = webhookIdAndEventTypeIdSchema.safeParse(rawInput);
+  if (!webhookIdAndEventTypeId.success) {
+    throw new TRPCError({ code: "PARSE_ERROR" });
+  }
```

Comment:
I think this could be simplified a bit by splitting it up into 2 base procedures.

- One can use [multiple input parsers](https://trpc.io/docs/v10/procedures#multiple-input-parsers) with the `webhookIdAndEventTypeIdSchema`
- One can be without

Reply:
Would you mind looking at that?

I don't really understand how I'd split this up since their endpoints mostly use optional input and putting the multiple input parsers would make all of these "required", i.e. not using the same middleware but early returning if the input isn't defined?

On the one hand, if they don't pass any input it's just an authedProcedure, but looks like even the `webhook.list` optionally use the input (which should in that case go through the middleware)

Reply:
It's an non-need optimization, let's get this merged first and then I can bring some ideas to the team on how to improve stuff! :)